monitor/account.go

```diff
@@ -0,0 +1,74 @@
+package monitor
+
+import (
+	"context"
+
+	"github.com/celo-org/kliento/contracts"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/log"
+)
+
+type accountProcessor struct {
+	ctx      context.Context
+	logger   log.Logger
+	address  common.Address
+	accounts *contracts.Accounts
+}
+
+func NewAccountsProcessor(ctx context.Context, logger log.Logger, address common.Address, accounts *contracts.Accounts) *accountProcessor {
+	return &accountProcessor{
+		ctx:      ctx,
+		logger:   logger,
+		address:  address,
+		accounts: accounts,
+	}
+}
+
+func (p accountProcessor) ObserveState() {
+	return
+}
+
+func (p accountProcessor) HandleLog(eventLog *types.Log) {
+	logger := p.logger.New("contract", "Accounts", "logIndex", eventLog.Index)
+	if eventLog.Address == p.address {
+		eventName, eventRaw, ok, err := p.accounts.TryParseLog(*eventLog)
+		if err != nil {
+			logger.Warn("Ignoring event: Error parsing accounts event", "err", err, "eventId", eventLog.Topics[0].Hex())
+			return
+		}
+		if !ok {
+			return
+		}
+
+		var params = make([]interface{}, 0)
+		params = append(params, "eventName", eventName)
+
+		switch e := eventRaw.(type) {
+		case *contracts.AccountsAttestationSignerAuthorized:
+			params = append(params, "account", e.Account, "signer", e.Signer)
+		case *contracts.AccountsVoteSignerAuthorized:
+			params = append(params, "account", e.Account, "signer", e.Signer)
+		case *contracts.AccountsValidatorSignerAuthorized:
+			params = append(params, "account", e.Account, "signer", e.Signer)
+		case *contracts.AccountsAttestationSignerRemoved:
+			params = append(params, "account", e.Account, "oldSigner", e.OldSigner)
```

Comment:
What do you think about still naming this key `signer`? This feels more like a foreign key that is common amonge these events

Reply:
Yeah makes sense. Probably easier from a querying perspective as well